executor/api/rabbitmq/rabbitmq_client.go

```diff
@@ -0,0 +1,145 @@
+package rabbitmq
+
+import (
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/wagslane/go-rabbitmq"
+)
+
+const (
+	publishMandatory = true
+	publishImmediate = false
+)
+
+type RabbitMqConnection struct {
+	conn *rabbitmq.Conn
+	log  logr.Logger
+}
+
+type RabbitMqConsumer struct {
+	rabbitmq.Consumer
+}
+
+type RabbitMqPublisher struct {
+	rabbitmq.Publisher
+	log logr.Logger
+}
+
+func createRabbitMQConnection(brokerUrl string, log logr.Logger) (ConnectionWrapper, error) {
```

Comment:
I presume this accepts multiple hosts in brokerUrl?

Reply:
if a host goes down will this seamlessly connect to another host?

Reply:
i do not think it accepts multiple hosts.  But i will test the failover to another host via DNS resolution/service health

Reply:
how does this client support auto reconnect?

Reply:
the consumer/publisher themselves use a connection from the wagslane client that reconnects if it disconnects